src/dysh/spectra/tests/test_scan.py

```diff
@@ -183,6 +180,85 @@ def test_compare_with_GBTIDL(self, data_dir):
         # the difference at the ~0.06 K level
         assert np.nanmedian(ratio) <= 0.998
 
+    def test_synth_spectra(self, data_dir):
+        """Test subbeamnod using synthithic spectra."""
+        # Load a file with subbeamnod observations.
+        sdf_file = f"{data_dir}/TRCO_230413_Ka/TRCO_230413_Ka_scan43.fits"
+        sdf = gbtfitsload.GBTFITSLoad(sdf_file)
+
+        # Generate fake data.
+        def gauss(x, a, s, c):
+            return a * np.exp(-((x - c) ** 2.0) / (2 * s**2.0))
+
+        # Make a fake data set.
+        gain = 1e8
+        tsys = 100
+        tcont = 10
+        tcal = sdf["TCAL"][0]
+        a = 10
+        s = 5
+        c = 800
+        x = np.arange(0, sdf["DATA"].shape[1], 1)
+        nchan = sdf["DATA"].shape[1]
+
+        # Signal and reference integrations.
+        sig_mask = sdf["SUBREF_STATE"].to_numpy() == -1
+        ref_mask = sdf["SUBREF_STATE"].to_numpy() == 1
+        n_sig = sig_mask.sum()
+        n_ref = ref_mask.sum()
+
+        # Generate astro signal.
+        np.random.seed(0)  # Fix the noise.
+        sig_mod = np.random.normal(loc=0, scale=5, size=(n_sig, nchan)) + gauss(x, a, s, c) + tcont
+        ref_mod = np.random.normal(loc=0, scale=5, size=(n_ref, nchan))
+
+        # Add noise diode.
+        sig_cal_on = sdf["CAL"].to_numpy()[sig_mask] == "T"
+        sig_mod[sig_cal_on] += tcal
+        ref_cal_on = sdf["CAL"].to_numpy()[ref_mask] == "T"
+        ref_mod[ref_cal_on] += tcal
+
+        # Convert to counts.
+        p_sig_mod = gain * (sig_mod + tsys)
+        p_ref_mod = gain * (ref_mod + tsys)
+
+        # Replace data.
+        sdf["TCAL"] = tcal
+        new_data = np.empty_like(sdf["DATA"])
+        new_data[sig_mask] = p_sig_mod
+        new_data[ref_mask] = p_ref_mod
+        sdf["DATA"] = new_data
```

Comment:
I expect this generates a warning.   You could filter the warning beforehand.  Or not.

Reply:
Good idea.